### ISSUE TITLE
Need clarification on survey and survey responses models

### ISSUE THREAD

**Issue by msyao-github** [ga-wdi-boston/group-project#92]
We want to call **question** attribute from **surveys** in **survey responses** model -- and do we need to update the survey responses controller with that change? 

```javascript
questions: {
    text: {
      type: String,
      required: false
    }
  }
```

We are not sure how to google the answer. We know we need to update response in the model--but we are not sure how to change.
Currently in our surveyresponse model:

```javascript
'use strict';

const mongoose = require('mongoose');

const surveyResponsesSchema = new mongoose.Schema({
  response: {
    type: String,
    required: true,
  },
  _survey:{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Survey',
        required: true,

  },
  _owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: false,
  },
}, {
  timestamps: true,
  toJSON: { virtuals: true },
});

const SurveyResponses = mongoose.model('SurveyResponses', surveyResponsesSchema);

module.exports = SurveyResponses;
```

Our survey model:

```javascript
'use strict';

const mongoose = require('mongoose');

const surveySchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  description: {
    type: String,
  },
  questions: {
    text: {
      type: String,
      required: false
    }
  },
  _owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
}, {
  timestamps: true,
  toJSON: { virtuals: true },
});

// surveySchema.virtual('length').get(function length() {
//   return this.text.length;
// });

const Survey = mongoose.model('Survey', surveySchema);

module.exports = Survey;
```

**Comment by msyao-github** [comment]
@erdesi90562 @Knguyen21

**Comment by jrhorn424** [comment]
It isn't clear what you want to do. Could you ask your question in a different way? Can you describe example output, and tell me what controller action and route you're thinking of using?

Here's a few things I think might need to change based on our conversation from yesterday:

```diff
- questions: {
+ questions: [

    text: {
      type: String,
      required: false
    }
-  }
+  ]
```

```diff
const surveyResponsesSchema = new mongoose.Schema({
-  response: {
-    type: String,
-    required: true,
-  },
  _survey:{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Survey',
        required: true,

  },
  _owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: false,
  },
}, {
  timestamps: true,
  toJSON: { virtuals: true },
});
```

You can't know what the full schema is going to be for surveyResponse, since the keys are going to be the values from question texts.

**Comment by jrhorn424** [comment]
@msyao-github If you don't mind, record a few notes about what you did with survey responses and close. If you want to talk about this further, feel free to leave it open, though.

**Comment by msyao-github** [comment]
Hi Jeff,

We associated survey responses to surveys through survey id in the survey response model. We also added routes so that survey responses can be created and show in association with survey id.

```javascript
routes.get('/surveys/:id/surveyResponses', 'surveyResponses#index');
routes.post('/surveys/:id/surveyResponses', 'surveyResponses#create');
``` 
Because in our survey web app, we only have one question and one answer per survey, we didn't need to make additional model.

Thank you!

cc: 
@erdesi90562 @Knguyen21
